test/cpp/tensorexpr/test_simplify.cpp

```diff
@@ -4824,7 +4824,303 @@ TEST(Simplify, SimplifyBroadcastTermExpander) {
   }
 }
 
-TEST(Simplify, DISABLED_CompareSelectCondAlwaysInLoopBounds) {
+TEST(Simplify, CompareSelectLoopBounds) {
+  constexpr int N = 8;
```

Comment:
@navahgar do we need to test dynamic shapes here? 
```
VarPtr N = alloc<Var>("N", kInt);
```

Reply:
Yes, it will be good to have some tests with dynamic shapes, to ensure that nothing breaks with these changes.